Fix `make test-style` warnings

Most of these are very easy to fix.

```console
⇒  make test-style
Running golint...
cmd/helm/helm.go:81:1: exported function NewClient should have comment or be unexported
pkg/client/client.go:35:1: comment on exported var DefaultHTTPTimeout should be of the form "DefaultHTTPTimeout ..."
pkg/client/client.go:38:1: comment on exported var DefaultHTTPProtocol should be of the form "DefaultHTTPProtocol ..."
pkg/registry/registryprovider.go:37:6: type name will be used as registry.RegistryProvider by other packages, and that stutters; consider calling this Provider
pkg/registry/registryprovider.go:283:1: comment on exported var TemplateRegistryMatcher should be of the form "TemplateRegistryMatcher ..."
pkg/registry/registryprovider.go:286:1: comment on exported var PackageRegistryMatcher should be of the form "PackageRegistryMatcher ..."
pkg/registry/registryprovider.go:289:1: comment on exported var GCSRegistryMatcher should be of the form "GCSRegistryMatcher ..."
pkg/util/httputil.go:117:1: exported function LogHandlerText should have comment or be unexported
```

--- a/cmd/helm/helm.go
+++ b/cmd/helm/helm.go
@@ -78,6 +78,7 @@ func run(c *cli.Context, f func(c *cli.Context) error) {
 	}
 }
 
+// NewClient creates a new client instance preconfigured for CLI usage.
 func NewClient(c *cli.Context) *client.Client {
 	host := c.GlobalString("host")
 	debug := c.GlobalBool("debug")

--- a/cmd/resourcifier/main.go
+++ b/cmd/resourcifier/main.go
@@ -17,9 +17,6 @@ limitations under the License.
 package main
 
 import (
-	"github.com/kubernetes/deployment-manager/pkg/util"
-	"github.com/kubernetes/deployment-manager/pkg/version"
-
 	"flag"
 	"fmt"
 	"log"
@@ -28,6 +25,8 @@ import (
 
 	"github.com/gorilla/handlers"
 	"github.com/gorilla/mux"
+	"github.com/kubernetes/deployment-manager/pkg/util"
+	"github.com/kubernetes/deployment-manager/pkg/version"
 )
 
 // Route defines a routing table entry to be registered with gorilla/mux.

--- a/pkg/client/client.go
+++ b/pkg/client/client.go
@@ -32,10 +32,10 @@ import (
 	"github.com/kubernetes/deployment-manager/pkg/common"
 )
 
-// The default HTTP timeout
+// DefaultHTTPTimeout is the default HTTP timeout.
 var DefaultHTTPTimeout = time.Second * 10
 
-// The default HTTP Protocol
+// DefaultHTTPProtocol is the default HTTP Protocol (http, https).
 var DefaultHTTPProtocol = "http"
 
 // Client is a DM client.

--- a/pkg/registry/registryprovider.go
+++ b/pkg/registry/registryprovider.go
@@ -280,13 +280,13 @@ func (gcsrp gcsRegistryProvider) createGCSClient(credentialName string) (*http.C
 	return config.Client(oauth2.NoContext), nil
 }
 
-// RE for a registry type that does support versions and has collections.
+// TemplateRegistryMatcher is an RE for a registry type that does support versions and has collections.
 var TemplateRegistryMatcher = regexp.MustCompile("github.com/(.*)/(.*)/(.*)/(.*):(.*)")
 
-// RE for a registry type that does not support versions and does not have collections.
+// PackageRegistryMatcher is an RE for a registry type that does not support versions and does not have collections.
 var PackageRegistryMatcher = regexp.MustCompile("github.com/(.*)/(.*)/(.*)")
 
-// RE for GCS storage
+// GCSRegistryMatcher is an RE for GCS storage
 var GCSRegistryMatcher = regexp.MustCompile("gs://(.*)/(.*)")
 
 // IsGithubShortType returns whether a given type is a type description in a short format to a github repository type.

--- a/pkg/util/httputil.go
+++ b/pkg/util/httputil.go
@@ -114,6 +114,7 @@ func TestHandlerWithURL(handler http.Handler, method, urlString string) (*httpte
 	return NewHandlerTester(handler).TestWithURL(method, urlString)
 }
 
+// LogHandlerText logs a line of text for a handler.
 func LogHandlerText(handler string, v string) {
 	log.Printf("%s: %s\n", handler, v)
 }

